src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -47,6 +47,41 @@ internal static ZipkinSpan ToZipkinSpan(this Activity activity, ZipkinEndpoint l
 
             activity.EnumerateTags(ref tagState);
 
+            // When status is set on Activity using the native Status field in activity,
+            // which was first introduced in System.Diagnostic.DiagnosticSource 6.0.0.
+            if (activity.Status != ActivityStatusCode.Unset)
+            {
+                if (activity.Status == ActivityStatusCode.Ok)
+                {
+                    PooledList<KeyValuePair<string, object>>.Add(
+                    ref tagState.Tags,
+                    new KeyValuePair<string, object>(
+                        SpanAttributeConstants.StatusCodeKey,
+                        "OK"));
+                }
+
+                // activity.Status is Error
+                else
+                {
+                    PooledList<KeyValuePair<string, object>>.Add(
```

Comment:
populate error, "description" also right here, so you don't have to check if Status is error again.